source/material_model/grain_size.cc

```diff
@@ -393,70 +398,38 @@ namespace aspect
-      const double diff_viscosity = diffusion_viscosity(temperature,pressure,composition,strain_rate,position) ;
-
-      // Start the iteration with the full strain rate
-      double dis_viscosity;
-      if (viscosity_guess == 0)
-        dis_viscosity = dislocation_viscosity_fixed_strain_rate(temperature,pressure,std::vector<double>(),strain_rate,position);
-      else
-        dis_viscosity = viscosity_guess;
-
-      double dis_viscosity_old = 0;
-      unsigned int i = 0;
-      while ((std::abs((dis_viscosity-dis_viscosity_old) / dis_viscosity) > dislocation_viscosity_iteration_threshold)
-             && (i < dislocation_viscosity_iteration_number))
-        {
-          const SymmetricTensor<2,dim> dislocation_strain_rate = diff_viscosity
-                                                                 / (diff_viscosity + dis_viscosity) * strain_rate;
-          dis_viscosity_old = dis_viscosity;
-          dis_viscosity = dislocation_viscosity_fixed_strain_rate(temperature,
-                                                                  pressure,
-                                                                  std::vector<double>(),
-                                                                  dislocation_strain_rate,
-                                                                  position);
-          ++i;
-        }
-
-      Assert(i<dislocation_viscosity_iteration_number,ExcInternalError());
-
-      return dis_viscosity;
-    }
-
-
-
-    template <int dim>
-    double
-    GrainSize<dim>::
-    dislocation_viscosity_fixed_strain_rate (const double      temperature,
-                                             const double      pressure,
-                                             const std::vector<double> &,
-                                             const SymmetricTensor<2,dim> &dislocation_strain_rate,
-                                             const Point<dim> &position) const
-    {
-      const SymmetricTensor<2,dim> shear_strain_rate = dislocation_strain_rate - 1./dim * trace(dislocation_strain_rate) * unit_symmetric_tensor<dim>();
-      const double second_strain_rate_invariant = std::sqrt(std::max(-second_invariant(shear_strain_rate), 0.));
-
-      // Currently this will never be called without adiabatic_conditions initialized, but just in case
+      const double adiabatic_temperature = this->get_adiabatic_conditions().is_initialized()
+                                           ?
+                                           this->get_adiabatic_conditions().temperature(position)
+                                           :
+                                           temperature;
       const double adiabatic_pressure = this->get_adiabatic_conditions().is_initialized()
                                         ?
                                         this->get_adiabatic_conditions().pressure(position)
                                         :
                                         pressure;
```

Comment:
Just a question: For the diffusion viscosity, you compute this outside the function call. Here, you compute it inside. 
Wouldn't it make more sense to do it the same way for both functions? (Both in terms of speed and readability?)